File: app/types/export.ts
export type ExportFormat = 'txt' | 'markdown' | 'html' | 'project_package'
export type ExportRange = 'all' | 'current' | 'selected'

export interface ExportDocumentsInput {
    projectId: string
    format: ExportFormat
    range?: ExportRange | null
    documentId?: string | null
    documentIds?: string[] | null
    outputPath?: string | null
}

export interface ExportProjectInput {
    projectId: string
    format?: ExportFormat | null
    outputPath?: string | null
}

export interface ExportResult {
    path: string
    format: ExportFormat
    documentCount: number
    characterCount: number
    createdAt: number
}

export interface ImportTextFileInput {
    projectId: string
    path: string
    format?: 'txt' | 'markdown' | 'html' | null
    parentId?: string | null
    title?: string | null
}

export interface ImportResult {
    documentId: string
    title: string
    characterCount: number
    importedAt: number
}

export interface BackupItem {
    id: string
    projectId: string
    path: string
    sizeBytes: number
    createdAt: number
}

File: src-tauri/src/models/export.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExportDocumentsInput {
    pub project_id: String,
    pub format: String,
    pub range: Option<String>,
    pub document_id: Option<String>,
    pub document_ids: Option<Vec<String>>,
    pub output_path: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExportProjectInput {
    pub project_id: String,
    pub format: Option<String>,
    pub output_path: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ExportResultDto {
    pub path: String,
    pub format: String,
    pub document_count: i64,
    pub character_count: i64,
    pub created_at: i64,
}
